cmp/form/FormModel.ts

```diff
@@ -274,4 +310,50 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToInclude = null,
+        fieldsToExclude = null,
+        path = 'formValues',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const allFields = Object.keys(this.fields);
+        const fieldNamesToPersist = fieldsToExclude
+            ? without(allFields, ...fieldsToExclude)
+            : (fieldsToInclude ?? allFields);
+
+        PersistenceProvider.create({
+            persistOptions: {
+                path,
+                ...rootPersistWith
+            },
+            target: {
+                getPersistableState: () =>
+                    new PersistableState(this.serialize(pick(this.getData(), fieldNamesToPersist))),
+                setPersistableState: ({value: formValues}) =>
+                    this.setValues(this.deserialize(pick(formValues, fieldNamesToPersist)))
+            },
+            owner: this
+        });
+    }
+
+    private serialize = (formValue: unknown) => {
+        if (isArray(formValue)) return formValue.map(this.serialize);
+        if (isDate(formValue)) return {__type: 'date', value: formValue.toJSON()};
+        if (isLocalDate(formValue)) return {__type: 'localDate', value: formValue.toJSON()};
```

Comment:
I'm down with this use of `__type` - seems very standard and reasonable to me. Anyone inspecting the persisted contents should be able to suss out what's going on.

In a few other places we use `_xh` as a prefix - e.g. `_xhType` or `_xhFieldType`.  Mainly thinking of possible future cases where we are serializing type/value pairs in other contexts and this convention gets spread around, would there be any advantage to using that `xh` to signal "this is something that Hoist is processing"?  

Don't feel strongly about it, just throwing it out there. Am 100% fine if `__type` is a more general convention, or feels like one.

Great that we are pushing past this trickiness.

Reply:
I had the same thought last night and agree -- probably better to use the xh prefix so we could do this elsewhere, be consistent, and not worry about a name collision.

Was thinking it would be nice if our `markPersist` and `@persist` decorators also handled serializing / deserializing `Dates` and `LocalDates` and collections that contain them. (Ex. `bindable` date input that you want to persist / don't need a whole `Form` and `FormModel`).

Reply:
Just to clarify -- not saying we need to add that for this PR, but using the `xh` prefix convention gives us room to implement this in more places where we might not be able to make as many assumptions about the underlying data.

Reply:
Great agreed 100% good point about persisting (Local)Date properties directly - as a follow-on enhancement